src/orderbook/state.ts

```diff
@@ -101,23 +172,82 @@ export class AccountState {
             validUntil: parseInt(data.validUntil),
             priceNumerator: BigInt(data.priceNumerator),
             priceDenominator: BigInt(data.priceDenominator),
+            remainingAmount: BigInt(data.priceDenominator),
           })
           break
+        case 'SolutionSubmission':
+          /* eslint-disable no-case-declarations */
+          const submitter = data.submitter
+          const burntFees = BigInt(data.burntFees)
+          this.updateBalance(submitter, 0, amount => amount + burntFees)
+          lastSolution = { submitter, burntFees }
```

Comment:
Same comment as before, let's extract those into helper methods to keep this giant switch statement dump.